app/actors/ClientActor.java

```diff
@@ -1,78 +1,80 @@
 package actors;
 
+import akka.actor.AbstractActor;
 import akka.actor.ActorRef;
 import akka.actor.Props;
-import akka.actor.UntypedActor;
-import akka.japi.Procedure;
+import akka.event.Logging;
+import akka.event.LoggingAdapter;;
+import akka.japi.pf.ReceiveBuilder;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import play.Logger;
 import play.libs.Json;
+import scala.PartialFunction;
+import scala.runtime.BoxedUnit;
 import utils.ProtocolCodec;
+
 import java.util.Optional;
 import java.util.Random;
 
-public class ClientActor extends UntypedActor {
-    public static Props props(ActorRef out) {
-        return Props.create(ClientActor.class, out);
-    }
+public class ClientActor extends AbstractActor {
```

Comment:
We are using  Akka actors with java lambda support and the API in this case extends the AbstractActor